explanation/maintenance-mode.rst

```diff
@@ -0,0 +1,65 @@
+Maintenance Mode
+================
+
+Overview
+--------
+
+Maintenance mode provides a safeguard for operators performing critical operations 
+that could potentially disrupt services or result in data loss. Before executing 
+any high-risk actions on a node, operators should enable maintenance mode to 
+minimize impact and ensure system stability.
+
+States of the Node
+------------------
+
+A node in Sunbeam can exist in one of three states: **Active**, **Maintenance**, or **Decommissioned**. 
+
+- **Active**: The node is fully operational, healthy, and actively contributing to the cluster.  
+- **Maintenance**: The node remains part of the cluster, but safeguards are in place to ensure that its temporary unavailability does not impact overall system stability. Necessary precautions, such as disabling services or redistributing workloads, have been applied to make the node safe for maintenance operations. Once the maintenance operation is complete, the operator should disable maintenance mode to return the node to the **Active** state.  
+- **Decommissioned**: The node is no longer part of the cluster and does not participate in cluster activities. A **Decommissioned** node does not impact cluster operations, as its responsibilities have already been reassigned.  
+
+By defining these states, Sunbeam ensures that operators can perform maintenance efficiently while minimizing disruptions to the system.
+
+Definition of Maintenance Mode in Sunbeam
+-----------------------------------------
+
+In the Sunbeam architecture, each node in a deployment can assume one or more roles. 
+When maintenance mode is enabled, Sunbeam ensures that all roles on the node reach 
+a maintainable state before proceeding.
+
+The following sections define the maintenance mode status for each role:
+
+Compute Role
+~~~~~~~~~~~~
+
+- Nova compute status is set to *disabled*, preventing new instances from being scheduled on the node.
+- No instances are running on the node, ensuring that workloads are not affected during maintenance operations.
+- Hypervisor services remain active.
+
+Storage Role
+~~~~~~~~~~~~
+
+- OSD (Object Storage Daemon):
+    - OSDs on the node must be ``ok-to-stop``, ensuring sufficient redundancy to tolerate the loss of OSDs on the node. See ``ok-to-stop`` part in `ceph administration tool`_ for more information.
+    - *(Optional)* OSDs on the node must be either:
```

Comment:
Not sure about `OUT and DOWN` part; stopping the osd service will only make the OSDs on the node `DOWN`, but probably not `OUT`